iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheck.java

```diff
@@ -20,91 +20,100 @@
-        return;
-      }
+  private static boolean isContainersAbsent(BlockObject spec) {
+    return spec.attribute("containers").tree == null;
+  }
+
+  private void checkAccountSecurity(BlockObject document, BlockObject spec, String namespace, @Nullable String accountName) {
+    // If we can find any RoleBinding or ClusterRoleBinding that references the service account, it's compliant and we stop there
+    if (hasAnyBondedRole(document, namespace, accountName)) {
+      return;
+    }
+
+    // If we can find a ServiceAccount of that name with automountServiceAccountToken set to false, it's compliant and we stop there
+    var serviceAccounts = findGlobalResources(ServiceAccount.class, namespace, document).stream()
+      .filter(serviceAccount -> serviceAccount.name().equals(accountName))
+      .toList();
+    if (serviceAccounts.stream().anyMatch(serviceAccount -> serviceAccount.automountServiceAccountToken().isFalse())) {
+      return;
+    }
 
-      List<ServiceAccount> linkedServiceAccounts = retrieveLinkedServiceAccount(blockObject);
-      if (linkedServiceAccounts.isEmpty()) {
-        specAsAttributeObject.reportOnKey(message);
+    // If we reach there, then we couldn't find any security measure in place -> raise an issue
+    if (serviceAccounts.isEmpty()) {
+      if (accountName == null) {
+        spec.attribute("containers")
+          .reportOnKey(MESSAGE_BIND_ACCOUNT_NAME);
       } else {
-        boolean hasAtLeastOneAccountCompliant = linkedServiceAccounts.stream().anyMatch(account -> account.automountServiceAccountToken().isFalse());
-        if (!hasAtLeastOneAccountCompliant) {
-          // report on first linked account - there should be only one
-          reportIssueWithLinkedAccount(linkedServiceAccounts.get(0), blockObject, message);
-        }
+        spec.attribute(SERVICE_ACCOUNT_NAME)
+          .reportOnValue(MESSAGE_BIND_ACCOUNT_RESOURCE);
       }
+    } else {
+      // report on first account - there should be only one
+      reportIssueWithLinkedAccount(spec.attribute(SERVICE_ACCOUNT_NAME), serviceAccounts.iterator().next());
     }
   }
 
-  private static void reportIssueWithLinkedAccount(ServiceAccount linkedAccount, BlockObject blockObject, String message) {
-    var blockSpec = blockObject.block("spec");
-    var serviceAccountNameAttr = blockSpec.attribute("serviceAccountName");
-    List<SecondaryLocation> secondaryLocations = new ArrayList<>();
-    secondaryLocations.add(new SecondaryLocation(serviceAccountNameAttr.tree.value(), "Through this service account"));
-    if (linkedAccount.automountServiceAccountToken().isTrue()) {
-      secondaryLocations.add(new SecondaryLocation(linkedAccount.valueLocation(), "Change this setting", linkedAccount.filePath()));
-    }
-    blockObject.attribute("spec").reportOnKey(message, secondaryLocations);
+  private boolean hasAnyBondedRole(BlockObject document, String namespace, @Nullable String accountName) {
```

Comment:
```suggestion
  private boolean hasAnyBoundRole(BlockObject document, String namespace, @Nullable String accountName) {
```
I believe it would sound better :)